addons/portal/models/ir_ui_view.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+from odoo.http import request
+from odoo.addons.http_routing.models.ir_http import url_for
+
+
+class View(models.Model):
+    _inherit = "ir.ui.view"
+
+    @api.model
+    def _prepare_qcontext(self):
+        """ Returns the qcontext : rendering context with portal specific value (required
+            to render portal layout template)
+        """
+        qcontext = super(View, self)._prepare_qcontext()
+        if request and getattr(request, 'is_frontend', False):
+            Lang = request.env['res.lang']
+            portal_lang_code = request.env['ir.http']._get_frontend_langs()
+            qcontext.update(dict(
+                self._context.copy(),
```

Comment:
This is wrong cc @Gorash

Reply:
@Julien00859 Hello
Are you expecting us to do something here? `This is wrong` on a 17 months old PR is a bit light to expect anything.

Reply:
Indeed, it seems bad to me to put the whole context directly in the qcontext. This can give strange side effects. Like a key-value in the ´context` but which shouldn't be in the template *values*, a `t-if` using that key will not react correctly.
In addition `env` is already available and therefore nothing prevents to make `env.context` in the template.

Reply:
If you need to access a `key` from the environment context you should use `env.context["key"]`. The environment context variables should **not** be directly accessible as if they were global variables. This lead to poor code in templates and hard to debug bugs. Notably it breaks my httpocalypse because some keys have been removed from the context but where still used in some templates.

Reply:
You 2 realizes this line is here since mathusalem, or at least 2013? I did not get further because it is pointless but just to get an idea..
Lot of people rewrote/moved that line including @Gorash 
If it is needed to fix it, could you please do it?

[0e82448ed8](https://github.com/odoo-dev/odoo/commit/0e82448ed8adc54a73fd01046e7e5c4225b2bdd4)
[01a28b6e03](https://github.com/odoo-dev/odoo/commit/01a28b6e0348cc1714c2a27d4e2a5569518745f3)
[03606aa347](https://github.com/odoo-dev/odoo/commit/03606aa347f90333136e5a5fe35d17bc72a9cf44)
[b112d5d2f1](https://github.com/odoo-dev/odoo/commit/b112d5d2f1e8a6b4d457b0693f213a2e35510cfb)
[ba827fc6ee](https://github.com/odoo-dev/odoo/commit/ba827fc6ee4f450b9e3b02b07fe2966d9ecd5228)
[8c559044a5](https://github.com/odoo-dev/odoo/commit/8c559044a5e1be111dcb577ad7e5ab462c1002a0)
[fd09ddb6f3](https://github.com/odoo-dev/odoo/commit/fd09ddb6f3c40e9ec5854cc58c7eb791d4498dc0)
[62de570395](https://github.com/odoo-dev/odoo/commit/62de570395c06f5a17cdbd4d19c468c5a09641e4)

Reply:
@rdeodoo Julien found a bug due to this by refactoring ir.http. He indicates that the behavior is not good. I think he is right and that it is a source of problem and that we have left aside before. It will have to be changed one day (we are not ready for one or two years :-)) ). If this part of the code changes, do not redo the error.